lib/grppccom.gi

```diff
@@ -1207,15 +1207,51 @@ end);
 #M  ComplementcClassesRepresentatives( <G>, <N> )
 ##
 InstallMethod( ComplementClassesRepresentatives,
-  "tell that the normal subgroup must be solvable",IsIdenticalObj,
-  [IsGroup,IsGroup],-2*RankFilter(IsGroup),
+  "tell that the normal subgroup or factor must be solvable", IsIdenticalObj,
+  [ IsGroup, IsGroup ], -2*RankFilter(IsGroup),
 function( G, N )
-  if IsSolvableGroup(N) then
+  if IsSolvableGroup(N) or HasSolvableFactorGroup(G, N) then
     TryNextMethod();
   fi;
-  Error("Cannot compute complement classes for nonsolvable normal subgroups");
+  Error("Cannot compute complements if both N and G/N are nonsolvable");
 end);
```

Comment:
I personally do not mind one way or another. @hulpke put the original error message in the code, I just updated it to better reflect the current situation. Anyway, just let me know what to do with it.